src/definitions/extendType.ts

```diff
@@ -1,11 +1,14 @@
 import { assertValidName } from 'graphql'
 import { AllOutputTypesPossible } from '../typegenTypeHelpers'
 import { OutputDefinitionBlock } from './definitionBlocks'
+import { IsSubscriptionType, SubscriptionBuilder } from './subscriptionType'
 import { NexusTypes, withNexusSymbol } from './_types'
 
 export interface NexusExtendTypeConfig<TypeName extends string> {
   type: TypeName
-  definition(t: OutputDefinitionBlock<TypeName>): void
+  definition(
+    t: IsSubscriptionType<TypeName> extends true ? SubscriptionBuilder : OutputDefinitionBlock<TypeName>
```

Comment:
Make `extendType` builder API polymorphic over the value of `TypeName`. Specifically branching over if it is `Subscription` or not. This is needed since the types are different. The naming is a new problem now. That is, `OutputDefinitionBlock` should be called something else. Or, it should be refactored to still fulfill the role of generic handling of `Output`.